### PR TITLE
bonus episode: rearrange packages and cleaning

### PR DESCRIPTION
**Pull request by amanmdesai**
<details>
<summary><strong>Instructions</strong></summary>

Thanks for contributing! :heart:

Keep in mind that **lesson maintainers are volunteers** and it may take them some time to respond to your contribution.
To ensure that the right people get notified, you can tag some of the last contributors with `@githubname`.

Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum. If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact the HSF training convenors ([contacts here](https://hepsoftwarefoundation.org/workinggroups/training.html)).

You may delete these instructions from your pull request.

\- HSF Training
</details>

### CODE REVIEW

File: _episodes/08-bonus-episode.md [hunk 1]
```diff
@@ -84,38 +83,38 @@ It is possible to publish singularity images with GitHub packages. To do so, one
...
-           echo ${{ secrets.GITHUB_TOKEN }} | singularity remote login -u ${{ secrets.GHCR_USERNAME }} --password-stdin oras://ghcr.io
-           singularity push container.sif oras://ghcr.io/${GITHUB_REPOSITORY}:${tag}
+  build-test-container:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    container:
+        image: quay.io/singularity/singularity:v3.8.1
```
Comment (klieret) [inline _episodes/08-bonus-episode.md]: I'm not really sure what's going on here with the whitespace, isn't this mixing tabs and spaces here?

**Comment by klieret** [comment]
What's the reason behind rearranging the packages?

**Comment by amanmdesai** [comment]
> What's the reason behind rearranging the packages?

So I have removed package such as zfit. And rearranged them as per their use case. (e.g., numpy, uproot, like packages together, plotting libraries, etc.)

File: _episodes/08-bonus-episode.md (hunk 1 above)
Comment (amanmdesai) [reply _episodes/08-bonus-episode.md]: I am not sure about it. However, locally, the website renders the same as it should.

**Comment by amanmdesai** [comment]
Please do let me know if the packages can be further rearranged or if we need some more.

File: _episodes/08-bonus-episode.md (hunk 1 above)
Comment (klieret) [reply _episodes/08-bonus-episode.md]: can you check the file? I wouldn't use tabs.

**Comment by klieret** [comment]
Does having multiple `%post` sections instead of one actually add anything? Is it like docker layers that make it build faster if you only make changes in e.g., the last section?

Also pinging @michmx because I don't know much

**Comment by klieret** [comment]
Let me reassign this issue to @michmx

File: _episodes/08-bonus-episode.md (hunk 1 above)
Comment (amanmdesai) [reply _episodes/08-bonus-episode.md]: Thanks. I have removed the whitespaces.

**Comment by amanmdesai** [comment]
@klieret, @michmx , It does not seem to be advantageous (sorry I think I got it confused with multiple docker layers). 

I ran a few tests here: https://github.com/amanmdesai/test-singularity

The single post layer seems to be faster (2 min 26 seconds) and multiple layers takes (3 min 4 seconds). 
(Only counting the build time)

I think single layer would then be better as far as build time is concerned.

File: _episodes/08-bonus-episode.md [hunk 2]
```diff
@@ -112,10 +111,10 @@ jobs:
       - name: Login and Deploy Container
         run: |
            echo ${{ secrets.GITHUB_TOKEN }} | singularity remote login -u ${{ secrets.GHCR_USERNAME }} --password-stdin oras://ghcr.io
-           singularity push container.sif oras://ghcr.io/${GITHUB_REPOSITORY}:${tag}
+           singularity push container.sif oras://ghcr.io/${GITHUB_REPOSITORY}:${tag
```
Comment (michmx) [inline _episodes/08-bonus-episode.md]: This change is not introducing a bug with a missing `}` ?

Comment (amanmdesai) [reply _episodes/08-bonus-episode.md]: Thanks. It was erased by mistake.

Comment (amanmdesai) [reply _episodes/08-bonus-episode.md]: I have updated the missing }

**Review by michmx — APPROVED** [review]

**Pull request merged**